core/templates/dev/head/domain/question/QuestionUpdateService.js

```diff
@@ -66,11 +66,15 @@ oppia.factory('QuestionUpdateService', [
             question.setLanguageCode(oldLanguageCode);
           });
       },
-      setQuestionStateData: function(question, oldStateData, newStateData) {
+      setQuestionStateData: function(question, oldStateData, updateFunction) {
+        updateFunction();
+        var newStateData = question.getStateData();
         _applyPropertyChange(
           question, QUESTION_PROPERTY_QUESTION_STATE_DATA,
-          newStateData.toBackendDict(), oldStateData.toBackendDict(),
+          newStateData.toBackendDict(),
+          oldStateData.toBackendDict(),
           function(changeDict, question) {
+            updateFunction();
```

Comment:
Sorry, I think I am still confused. Why is updateFunction() being called twice (here and 7 lines above)?

Also /cc @aks681

Reply:
Yeah, I am not sure if that initial updateFunction() is required.
Also, if you are already taking newValue from question object, then why is question.setStateData() called again below? Isn't question already updated? I think there, updateFunction() might be enough.

Reply:
Oops, yeah the second updateFunction() shouldn't be there, removed.

I put a comment trying to explain what's going on. I know this isn't too pretty, so I'm open to ideas for improving this.